src/main/java/newbank/server/commands/LogoutCommand.java

```diff
@@ -0,0 +1,24 @@
+package newbank.server.commands;
+
+import newbank.server.Customer;
+import newbank.server.CustomerID;
+import newbank.server.NewBank;
+
+public class LogoutCommand extends Command{
+    private final CustomerID customer;
+
+    public LogoutCommand(final NewBank bank, final String[] tokens, CustomerID customer){
+        this.customer = customer;
+    }
+
+    @Override
+    public String execute() {
+
+        if (!isLoggedIn(customer)) {
+            return "You are not logged in.";
+        }
+
+        //create list and set to empty?
+        return "SUCCESS: You have been logged out.";
```

Comment:
We need to at least empty the customer's ID, i.e. do the opposite of the "isLoggedIn()" method to log the customer out. As it stands, the customer is still logged in.

Regards,
Jules

Reply:
Thanks for the feedback! Yes I was trying that yesterday but struggled to set customerID to empty.

atm , isloggedin is checking if empty but I want to set to "".

customer.setkey().clear()?
=null?

Reply:
`isLoggedIn` is calling `.isEmpty()` on a string instance, so you can set it to `""` (or `new String`) and `.isEmpty()` will be false.

Reply:
@claudist  setKey() takes an argument, if you do setKey("") then it will clear the key, I think. You can then test it on the command line to make sure that it has logged you out.

Kindest Regards,
Jules

Reply:
Thank you both!!! I've just psuhed 0 I tested that on the command line and works. After logging out I entered another command e.g. 'showmyaccounts' and request denied because was logged out:)

Reply:
Nice one, you still have the other comment above to address.

Regards,
Jules

Reply:
i tried catching up but then:

C:\Users\cstoc\IdeaProjects\newbank\src\main\java\newbank\server\NewBankClientHandler.java:67:56
java: unreported exception newbank.server.exceptions.CommandInvalidSyntaxException; must be caught or declared to be thrown

I'm literally in a meeting right now, so regrettably will not get that fixed before 12 :/

Reply:
@claudist  Updated your code, could you review and merge when you are happy?

Thanks and Regards,
Jules